Source/OIDAuthState.m

```diff
@@ -477,7 +487,7 @@ - (void)performActionWithFreshTokens:(OIDAuthStateAction)action
                  originalAuthorizationResponse:_lastAuthorizationResponse
                                       callback:^(OIDTokenResponse *_Nullable response,
                                                  NSError *_Nullable error) {
-    dispatch_async(dispatch_get_main_queue(), ^() {
+    dispatch_async(dispatchQueue, ^() {
```

Comment:
I'm thinking that the lines updating OIDAuthState should be in dispatch_get_main_queue(), and the "actionToProcess" block below in the user's dispatchQueue, do you agree @StevenEWright?